README.Rmd

```diff
@@ -97,4 +97,41 @@ Transformations are functions with a fixed signature.
 
 * `x` A named list of parameter values. Each list item contains a vector of parameter values of a single parameter.
 * `dict` An environment that can be accessed using the `$` operator. It can contains values that don't belong to any parameter but are important for transformations.
-* `tags` A list of the tags for each parameter. Each parameter can have a various tags indicating additional characteristics.
\ No newline at end of file
+* `tags` A list of the tags for each parameter. Each parameter can have a various tags indicating additional characteristics.
+
+
+### ParamTree
+```{r}
+  pt = ParamTree$fac(
+      ParamTree$dn(node = ParamCategorical$new(id = "model", values = c("SVM", "RF"))),
+      ParamTree$dn(node = ParamReal$new(id = "C", lower = 0, upper = 100), depend = list(id = "model", val = "SVM")),
+      ParamTree$dn(node = ParamCategorical$new(id = "kernel", values = c("rbf", "poly")), depend = list(id = "model", val = "SVM")),
+      ParamTree$dn(node = ParamReal$new(id = "gamma", lower = 0, upper = 100), depend = list(id = "kernel", val = "rbf")),
+      ParamTree$dn(node = ParamInt$new(id = "n", lower = 1L, upper = 10L), depend = list(id = "kernel", val = "poly"))
+      )
+  pt$sample()
```

Comment:
sample() for ParamTree should really work like sample for ParamSetFlat

Reply:
why?